plugins/copr.py

```diff
@@ -475,28 +477,36 @@ def _download_repo(self, project_name, repo_filename):
         api_path = "/coprs/{0}/repo/{1}/dnf.repo?arch={2}".format(project_name, short_chroot, arch)
 
         try:
-            f = self.base.urlopen(self.copr_url + api_path, mode='w+')
-        except IOError as e:
+            response = urlopen(self.copr_url + api_path)
             if os.path.exists(repo_filename):
                 os.remove(repo_filename)
-            if '404' in str(e):
-                try:
-                    res = urlopen(self.copr_url + "/coprs/" + project_name)
-                    status_code = res.getcode()
-                except HTTPError as e:
-                    status_code = e.getcode()
-                if str(status_code) != '404':
-                    raise dnf.exceptions.Error(_("This repository does not have"
-                                                 " any builds yet so you cannot enable it now."))
-                else:
-                    raise dnf.exceptions.Error(_("Such repository does not exist."))
-            raise
-
-        for line in f:
-            if re.match(r"\[copr:", line):
-                repo_filename = os.path.join(self.base.conf.get_reposdir,
-                                             "_" + line[1:-2] + ".repo")
-            break
+        except HTTPError as e:
```

Comment:
If I'm not mistaken this exception represents any non-successful HTTP error code. This means you need to check the actual HTTP code, the error handling below I think is only meant for 404 not found. I'm not entirely sure but any other code is just an unexpected error which you should just print like:
```
Error: Request to "http://localhost:2/coprs/testuser/testproject/repo/fedora-30/dnf.repo?arch=x86_64" failed:  STATUS_CODE - SATUS_MESSAGE
```